pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/IntermediateResultsBlock.java

```diff
@@ -311,16 +343,50 @@ private DataTable getResultDataTable()
       throws IOException {
     DataTableBuilder dataTableBuilder = DataTableFactory.getDataTableBuilder(_dataSchema);
     ColumnDataType[] storedColumnDataTypes = _dataSchema.getStoredColumnDataTypes();
+    int numColumns = _dataSchema.size();
     Iterator<Record> iterator = _table.iterator();
-    while (iterator.hasNext()) {
-      Record record = iterator.next();
-      dataTableBuilder.startRow();
-      int columnIndex = 0;
-      for (Object value : record.getValues()) {
-        setDataTableColumn(storedColumnDataTypes[columnIndex], dataTableBuilder, columnIndex, value);
-        columnIndex++;
+    RoaringBitmap[] nullBitmaps = null;
+    if (_isNullHandlingEnabled) {
+      nullBitmaps = new RoaringBitmap[numColumns];
+      Object[] colDefaultNullValues = new Object[numColumns];
+      for (int colId = 0; colId < numColumns; colId++) {
+        if (storedColumnDataTypes[colId] != ColumnDataType.OBJECT) {
+          colDefaultNullValues[colId] = FieldSpec.getDefaultNullValue(FieldSpec.FieldType.METRIC,
```

Comment:
Several data types are not supported as METRIC. We should allow setting `null` values in `setDataTableColumn()`

Reply:
Changed to:
```
  // Store a dummy value that is both a valid numeric, and a valid hex encoded value.
  String specialVal = "30";
  colDefaultNullValues[colId] = storedColumnDataTypes[colId].toDataType().convert(specialVal);
```

Setting it in `setDataTableColumn()` may hurt performance (need to check for null and type together).